doc/design/determinism_annotation.md

```diff
@@ -0,0 +1,58 @@
+# Determinism Annotations
+
+## Cert
+
+All functions are to be certified, hence there will be no label to indicate this.
+
+## Deterministic
+
+`\deterministic` indicates that the function is real-time safe, i.e. returns in a finite amount of
+time which is (somewhat) predictable.
+
+In particular this rules out.
+
+1. not using exceptions
+2. no dynamic allocations from an external allocator (new, malloc)
+3. non-blocking
+
+(TBD: Non blocking may still use blocking calls, there must be a return guarantee though.)
+
+Note: deterministic is the wrong wording and should be changed.
+
+## Nondeterminstic
+
+Will not be labeled on its own. Instead we have labels indicating the details.
+
+### Blocking
+
+`\blocking` indicates the code will block and needs o be unblocked externally. Say a mutex lock that
```

Comment:
@MatthiasKillat I agree that `\blocking` tag is useful. I would like to suggest that the function with `\blocking` tag mentions which blocking call is used. For example, If this function with  `\blocking` tag is calling the mutex lock function, the exact mutex lock function name can be mentioned in this doxygen.